### PR TITLE
New rule: classMemberNoDefaultAccess.

### PR DESCRIPTION
**Pull request by meros**
Disallows public access modifiers that are not explicit in the class.

### CODE REVIEW

**Comment by leeavital** [comment]
Can you add this rule to the README and `sample.tslint.json`?

**Comment by meros** [comment]
Sure thing.

**Comment by adidahiya** [comment]
Sorry this has taken so long to get reviewed. Can you please rebase these changes on the latest `master` branch? There are a couple changes I'd like you to pick up and incorporate into this code:

1. We now use `tsconfig.json` project files in this project. So you don't have to use reference paths anywhere. You do, however, have to keep the relevant config files up to date when you add new source files to the project. See `src/tsconfig.json`, `src/rules/tsconfig.json`, `test/tsconfig.json`
2. `utils.ts` now provides a function called `Lint.hasModifier()` which should be able to replace much of the code in this new rule walker.

File: src/rules/classMemberNoDefaultAccessRule.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    public visitPropertyDeclaration(node: ts.PropertyDeclaration): void {
+        this.checkModifiers(node, getModifiers(node.modifiers));
+        super.visitPropertyDeclaration(node);
+    }
+
+    private checkModifiers(node: ts.Node, current: IModifiers): void {
+        if (!this.followsRules(current)) {
+            var message = "Default access modifier on member/method not allowed";
```
Comment (adidahiya) [inline src/rules/classMemberNoDefaultAccessRule.ts]: change this to be a `public static FAILURE_STRING` property on the `Rule` class in this file -- similar to how other rule walkers are structured.

File: test/rules/classMemberNoDefaultAccessTest.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/// <reference path='../references.ts' />
+
+describe("<class-member-no-default-access>", () => {
+    it("disallow default access on member", () => {
+        var fileName = "rules/classmembernodefaultaccess.test.ts";
+        var ClassMemberNoDefaultAccess = Lint.Test.getRule("class-member-no-default-access");
```
Comment (adidahiya) [inline test/rules/classMemberNoDefaultAccessTest.ts]: `ClassMemberNoDefaultAccessRule`

File: test/rules/classMemberNoDefaultAccessTest.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        var actualFailures = Lint.Test.applyRuleOnFile(fileName, ClassMemberNoDefaultAccess, []);
+
+        Lint.Test.assertFailuresEqual(actualFailures, [
+            Lint.Test.createFailure(fileName, [8, 5], [8, 15],
+                "Default access modifier on member/method not allowed"),
```
Comment (adidahiya) [inline test/rules/classMemberNoDefaultAccessTest.ts]: after addressing the above comments, you should be able to reference this via `Rule.FAILURE_STRING`

File: test/rules/classMemberNoDefaultAccessTest.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/// <reference path='../references.ts' />
+
+describe("<class-member-no-default-access>", () => {
+    it("disallow default access on member", () => {
+        var fileName = "rules/classmembernodefaultaccess.test.ts";
```
Comment (adidahiya) [inline test/rules/classMemberNoDefaultAccessTest.ts]: use `const` instead of `var`

File: README.md [hunk 5]
```diff
@@ -127,6 +127,7 @@ A sample configuration file with all options is available [here](https://github.
   * `"arguments"` checks alignment of function call arguments.
   * `"statements"` checks alignment of statements.
 * `ban` bans the use of specific functions. Options are ["object", "function"] pairs that ban the use of object.function()
+* `class-member-no-default-access` enforces using explicit visibility on class members
```
Comment (ashwinr) [inline README.md]: this is too verbose. let's call this `member-access`. in general, we should be limiting rule name verbosity.

**Comment by adidahiya** [comment]
@meros I like this rule, and I want it in tslint :)

Will you have time to address the PR comments soon? Or can I take the code and make a new PR?

**Comment by ashwinr** [comment]
@adidahiya let's take this up when you get time and merge it in. also let's rename the rule to `member-no-default-access` to be consistent with `member-ordering`.

**Comment by adidahiya** [comment]
I think `member-access` is slightly better

**Comment by ashwinr** [comment]
What would the value be? Just a Boolean?

Sent from a phone

On Jul 22, 2015, at 09:20, Adi Dahiya <notifications@github.com<mailto:notifications@github.com>> wrote:


I think member-access is slightly better

**Comment by meros** [comment]
Hi guys, sorry I have been a little under the radar a while. I have been in vacation mode, and in sweden that is a non-negligible part of the year  :) I will look into this again soon, probably next week. 

I see no problem in the comments so far. About the name, member-access doesn't really give a hint on what setting the flag would give. I like member-no-default-access the best, it's longer but the behaviour is spot on with what the module enforces.

**Comment by meros** [comment]
Please see this pull request instead: https://github.com/palantir/tslint/pull/552

**Pull request closed without merging**